cli/gitleaks/gitleaks.py

```diff
@@ -0,0 +1,229 @@
+#
+# Copyright (C) 2023 Red Hat, Inc.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+import logging
+import os
+import shutil
+import subprocess as sp
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any
+
+import requests
+from simple_logger.logger import get_logger
+
+BUILD_ID_VAR: str = "BUILD_ID"
+ARTIFACT_DIR_VAR: str = "ARTIFACT_DIR"
+
+DEFAULT_SERVER_URL: str = "https://patterns.security.redhat.com"
+DEFAULT_OUTPUT_FILE: str = "gitleaks_report.json"
+DEFAULT_TOKEN_PATH: str = "/tmp/secrets/rh-patterns-server/access-token"
+
+logger = get_logger(name="gitleaks")
+
+
+class EmptyJobDir(Exception):
```

Comment:
Why are these classes? I would say, just use the same format for checking if things are set as I do in [`cli/objects/rule.py`](https://github.com/CSPI-QE/firewatch/blob/main/cli/objects/rule.py#L40)

Reply:
This is just a style preference and I have found that it helps with readability and maintainability, and I normally would extract the exception classes to a separate file. I'll  remove the classes and inline the log statements for better consistency.